src/librustc_mir/transform/const_prop.rs

```diff
@@ -516,7 +521,22 @@ impl<'a, 'mir, 'tcx> ConstPropagator<'a, 'mir, 'tcx> {
         ))
     }
 
-    fn replace_with_const(&self, rval: &mut Rvalue<'tcx>, value: Const<'tcx>, span: Span) {
+    fn operand_from_ref(&self, mplace: MPlaceTy<'tcx>, span: Span) -> Option<Operand<'tcx>> {
+        Some(Operand::Constant(Box::new(Constant {
+            span,
+            ty: mplace.layout.ty,
+            user_ty: None,
+            literal: self.tcx.mk_const(*ty::Const::from_scalar(
+                    self.tcx,
+                    Scalar::Ptr(mplace.to_ptr().ok()?),
+                    mplace.layout.ty,
+                    ))
+        }
+        )))
+    }
+
+    fn replace_with_const(
+        &mut self, rval: &mut Rvalue<'tcx>, value: Const<'tcx>, source_info: SourceInfo) {
```

Comment:
Use block indentation

```suggestion
        &mut self,
        rval: &mut Rvalue<'tcx>,
        value: Const<'tcx>,
        source_info: SourceInfo,
    ) {
```